polygeist/lib/Dialect/Polygeist/Utils/TransformUtils.cpp

```diff
@@ -129,22 +156,95 @@ LoopGuardBuilder::create(LoopLikeOpInterface loop) {
       });
 }
 
+void LoopGuardBuilder::guardLoop(RegionBranchOpInterface ifOp) const {
+  createThenBody(ifOp);
+  createElseBody(ifOp);
+  replaceUsesOfLoopReturnValues(loop, ifOp);
+}
+
 //===----------------------------------------------------------------------===//
 // SCFLoopGuardBuilder
 //===----------------------------------------------------------------------===//
 
-SCFLoopGuardBuilder::SCFLoopGuardBuilder(LoopLikeOpInterface loop)
-    : LoopGuardBuilder(), SCFLoopVersionBuilder(loop) {}
+void SCFLoopGuardBuilder::guardLoop() const {
```

Comment:
`LoopGuardBuilder` isn't derived from `IfThenElseBuilder` and so it has no access to `createIfOp`. If I make it derive from that abstract class then I won't be able to derive `SCFLoopGuardBuilder` from `SCFIfBuilder` because that class is derived from `LoopGuardBuilder`, and I need to do that because `SCFLoopGuardBuilder` need to implement the `createCondition` member function. 

Note that this member function calls `LoopGuardBuilder::guardLoop` already where the common code has been placed.